src/widget.py

```diff
@@ -4,16 +4,22 @@
 
 
 def mask_account_card(numbers: Any) -> Any:
-    """Создаем новую функцию, которая также макскирует номер, но теперь с названием счета (карты)"""
+    """Mask account or card number based on input type."""
     new_numbers = numbers.split(" ")
+
     if "Счет" in numbers:
-        return f"Счет {get_mask_account(numbers)}"
+        masked_account = get_mask_account(new_numbers[-1])
+        return f"Счет {masked_account}" if masked_account else "Счет "
     else:
-        return f"{' '.join(new_numbers[0:-1])} {get_mask_card_number(new_numbers[-1])}"
+        masked_card = get_mask_card_number(new_numbers[-1])
+        return masked_card if masked_card else "Карта "
 
 
 def get_date(date: Union[str]) -> Union[str, int]:
```

Comment:
Union тут не нужен - всегда возвращается строка
```suggestion
def get_date(date: str) -> str:
```